[CI] HoistRemoteEnrichTopNTests testTopNSortExpressionWithinRemoteEnrichAliasing failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #11541 / ubuntu-2404-aarch64_checkpart3_platform-support-arm](https://gradle-enterprise.elastic.co/s/xysmdgv3jmida)
- [elasticsearch-periodic-platform-support #11541 / almalinux-8-aarch64_checkpart3_platform-support-arm](https://gradle-enterprise.elastic.co/s/gcirtawazx6ru)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:test" --tests "org.elasticsearch.xpack.esql.optimizer.rules.logical.HoistRemoteEnrichTopNTests.testTopNSortExpressionWithinRemoteEnrichAliasing" -Dtests.seed=A29DBB097F39CB34 -Dtests.locale=seh-MZ -Dtests.timezone=Asia/Anadyr -Druntime.java=25
```

**Applicable branches:**
9.2

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.optimizer.rules.logical.HoistRemoteEnrichTopNTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testTopNSortExpressionWithinRemoteEnrichAliasing'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: a string starting with "$$order_by$1$0"
     but: was "$$order_by$1$1"
```

**Issue Reasons:**
- [9.2] 2 failures in test testTopNSortExpressionWithinRemoteEnrichAliasing (3.6% fail rate in 55 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/rules/logical/HoistRemoteEnrichTopNTests.java
+++ b/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/rules/logical/HoistRemoteEnrichTopNTests.java
@@ -252,7 +252,7 @@ public class HoistRemoteEnrichTopNTests extends AbstractLogicalPlanOptimizerTest
         assertFalse(topn.local());
         Order topNOrder = topn.order().get(1);
         NamedExpression expr = as(topNOrder.child(), NamedExpression.class);
-        assertThat(expr.name(), startsWith("$$order_by$1$0"));
+        assertThat(expr.name(), startsWith("$$order_by$1$"));
         var enrich = as(topn.child(), Enrich.class);
         assertThat(enrich.mode(), is(Enrich.Mode.REMOTE));
         var innerTopN = as(enrich.child(), TopN.class);

